File: 9.Longest Consecutive Sequence/Solution.cpp
#include <vector>
#include <unordered_set>
#include <iostream>
using namespace std;

class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> numSet(nums.begin(), nums.end());
        int longest = 0;
        
        for (int num : numSet) {
            if (!numSet.count(num - 1)) {  // Check sequence start
                int current = num;
                int streak = 1;
                
                while (numSet.count(current + 1)) {  // Expand sequence
                    current++;
                    streak++;
                }
                
                longest = max(longest, streak);
            }
        }
        
        return longest;
    }
};

// Example usage
int main() {
    Solution sol;
    vector<int> nums = {1, 0, 1, 2};
    cout << sol.longestConsecutive(nums);  // Output: 3
    return 0;
}

File: 9.Longest Consecutive Sequence/Solution.py
def longestConsecutive(nums):
    num_set = set(nums)
    longest = 0
    
    for num in num_set:
        if num - 1 not in num_set:  # Check if it's a sequence start
            current_num = num
            current_streak = 1
            
            while current_num + 1 in num_set:  # Expand the sequence
                current_num += 1
                current_streak += 1
            
            longest = max(longest, current_streak)
    
    return longest

# Example usage
nums = [100, 4, 200, 1, 3, 2]
print(longestConsecutive(nums))  # Output: 4
